File: backend/app/schemas/supplier.py
from __future__ import annotations
from pydantic import BaseModel, Field
from typing import Optional

class SupplierCreate(BaseModel):
    name: str = Field(min_length=2, max_length=255)
    description: Optional[str]=None

class SupplierOut(BaseModel):
    id: int
    name: str
    description: Optional[str]=None
    owner_id: int
    owner: Optional['UserBasic'] = None

    class Config:
        from_attributes = True

# Resolve forward references after UserBasic is defined
from app.schemas.user import UserBasic
SupplierOut.model_rebuild()

File: frontend/types/supplier.ts
import { User } from './user';

export interface Supplier {
  id: number;
  name: string;
  description?: string;
  owner_id: number;
  owner?: User;
}

export interface SupplierCreate {
  name: string;
  description?: string;
}

export interface SupplierOut {
  id: number;
  name: string;
  description?: string;
  owner_id: number;
  owner?: User;
}
